internal/runner/nomad_manager_test.go

```diff
@@ -373,6 +374,16 @@ func (s *ManagerTestSuite) TestOnAllocationAdded() {
 }
 
 func (s *ManagerTestSuite) TestOnAllocationStopped() {
+	s.Run("returns false for idle runner", func() {
```

Comment:
Shouldn't we test the case that `true` is returned (i.e., a runner gets deleted twice, or the runner was not found) as well?

Reply:
I added a test for the scenario that a runner was not found.
This test simulates the race condition behavior of the inactivity timer (with the Nomad event stream handling) that led to the requirement of returning this boolean. The inactivity timer removes the runner (shortly before an allocation rescheduling event was processed).

Reply:
Awesome, thanks!